From 45734409dd45e6b117f7f9f7e11047f67a7accc1 Mon Sep 17 00:00:00 2001
From: Mayank Grover <groverm@codeaurora.org>
Date: Wed, 20 Jun 2018 12:52:55 +0530
Subject: QcomModulePkg: Initialize data buffers to zero before using.

Change to initialize data buffers to zero before using, this is
required as uninitialized data buffer may cause an information leak
or unexpected behaviour.

Change-Id: I16cff000202158821e3c4731f9da6a2e054e9783
---
 QcomModulePkg/Library/FastbootLib/FastbootCmds.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/QcomModulePkg/Library/FastbootLib/FastbootCmds.c b/QcomModulePkg/Library/FastbootLib/FastbootCmds.c
index 9389173..40cb699 100755
--- a/QcomModulePkg/Library/FastbootLib/FastbootCmds.c
+++ b/QcomModulePkg/Library/FastbootLib/FastbootCmds.c
@@ -1783,6 +1783,8 @@ FastbootCmdsInit (VOID)
 		return Status;
 	}
 
+	/* Clear allocated buffer */
+	gBS->SetMem ((VOID *)FastBootBuffer, (MAX_BUFFER_SIZE * 2), 0x0);
 	FastbootCommandSetup( (void*) FastBootBuffer, MAX_BUFFER_SIZE);
 	return EFI_SUCCESS;
 }
-- 
cgit v1.1

